src/Libraries/Watch3DNodeModelsWpf/Watch3DNodeViewCustomization.cs

```diff
@@ -93,47 +95,63 @@ public void CustomizeView(Watch3D model, NodeView nodeView)
-            // model whenever its size is updated. 
-            // Updated from (Watch3d)View.SizeChanged to nodeView.SizeChanged - height 
+            // When user sizes a watch node, only view gets resized. The actual
+            // NodeModel does not get updated. This is where the view updates the
+            // model whenever its size is updated.
+            // Updated from (Watch3d)View.SizeChanged to nodeView.SizeChanged - height
             // and width should correspond to node model and not watch3Dview
             nodeView.SizeChanged += (sender, args) =>
 			    model.SetSize(args.NewSize.Width, args.NewSize.Height);
 
             // set WatchSize in model
-            watch3DView.View.SizeChanged += (sender, args) => 
+            watch3DView.View.SizeChanged += (sender, args) =>
 			    model.SetWatchSize(args.NewSize.Width, args.NewSize.Height);
 
             var mi = new MenuItem { Header = Resources.ZoomToFit };
             mi.Click += mi_Click;
 
             nodeView.MainContextMenu.Items.Add(mi);
 
-            var backgroundRect = new Rectangle
-            {
-                HorizontalAlignment = HorizontalAlignment.Stretch,
-                VerticalAlignment = VerticalAlignment.Stretch,
-                IsHitTestVisible = false,
-            };
-            var bc = new BrushConverter();
-            var strokeBrush = (Brush)bc.ConvertFrom("#313131");
-            backgroundRect.Stroke = strokeBrush;
-            backgroundRect.StrokeThickness = 1;
-            var backgroundBrush = new SolidColorBrush(Color.FromRgb(240, 240, 240));
-            backgroundRect.Fill = backgroundBrush;
+            nodeView.centralGrid.Children.Add(watch3DView);
+            Grid.SetRow(watch3DView, 1);
 
-            nodeView.PresentationGrid.Children.Add(backgroundRect);
-            nodeView.PresentationGrid.Children.Add(watch3DView);
-            nodeView.PresentationGrid.Visibility = Visibility.Visible;
 
             DataBridge.Instance.RegisterCallback(
                 model.GUID.ToString(),
                 obj =>
                     nodeView.Dispatcher.Invoke(
                         new Action<object>(RenderData),
                         DispatcherPriority.Render,
-                        obj));       
+                        obj));
+
+
+            //////////////////////////////////////////////
+            // Buttons for adding and removing ports
+            //////////////////////////////////////////////
+
+            WrapPanel wrapPanel = new WrapPanel
+            {
+                VerticalAlignment = VerticalAlignment.Top,
+                HorizontalAlignment = HorizontalAlignment.Center
+            };
+
+            wrapPanel.Children.Add(
+                new DynamoNodeButton(nodeView.ViewModel.NodeModel, "AddInPort")
+                {
+                    Content = "+",
+                    Style = (Style)SharedDictionaryManager.DynamoModernDictionary["AddRemoveButton"]
```

Comment:
is it possible to use nameof or some other method to have this fail at compile time if the key is changed?

Reply:
I am not sure if nameof is possible here. This part of the code is based on the NodeViewCustomization of the VariableInputNode and there the name "AddInPort" etc.. were actually hardcoded